providers-sdk/v1/recording/recording.go

```diff
@@ -135,6 +135,8 @@ func NewWithFile(path string, opts RecordingOptions) (llx.Recording, error) {
 				Path:            path,
 				prettyPrintJSON: opts.PrettyPrintJSON,
 				doNotSave:       opts.DoNotSave,
+				assets:          map[uint32]*Asset{},
+				Assets:          []*Asset{},
```

Comment:
nit: you don't really need to initialize the slice here, `nil` would also do the job